packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,362 @@
+  onsetDate: undefined,
+  onsetDateInputName: undefined,
+  onsetDateInputOnChange: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class OnsetPicker extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    const ageValues = OnsetUtil.onsetToAge(this.props.birthdate, this.props.onsetDate);
+
+    this.state = {
+      granularity: this.props.granularity,
+      precision: this.props.precision,
+      onsetDate: this.props.onsetDate ? this.props.onsetDate : moment(),
+      age: ageValues.age,
+      ageUnit: ageValues.ageUnit,
+    };
+
+    // This binding is necessary to make `this` work in the callback
+    this.changeGranularity = this.changeGranularity.bind(this);
+    this.changePrecision = this.changePrecision.bind(this);
+    this.changeAge = this.changeAge.bind(this);
+    this.changeAgeUnit = this.changeAgeUnit.bind(this);
+    this.changeYear = this.changeYear.bind(this);
+    this.changeMonth = this.changeMonth.bind(this);
+    this.changeDate = this.changeDate.bind(this);
+  }
+
+  /*
+   * Change state for granularity
+   */
+  changeGranularity(event) {
+    if (event.target.value === 'AGE') { // Calculate age values and update onsetDate to match age calculation
+      const ageValues = OnsetUtil.onsetToAge(this.props.birthdate, this.state.onsetDate);
+      const newDate = moment(this.props.birthdate).add(ageValues.age, ageValues.ageUnit);
+
+      this.setState({
+        age: ageValues.age,
+        ageUnit: ageValues.ageUnit,
+        onsetDate: newDate,
+      });
+
+      if (this.props.onsetDateInputOnChange) {
+        this.props.onsetDateInputOnChange(newDate);
+      }
+    }
+    this.setState({ granularity: event.target.value });
+
+    if (this.props.granularitySelectOnChange) {
+      this.props.granularitySelectOnChange(event.target.value);
+    }
+  }
+
+  /*
+   * Change state for precision
+   */
+  changePrecision(event) {
+    this.setState({ precision: event.target.value });
+
+    if (this.props.precisionSelectOnChange) {
+      this.props.precisionSelectOnChange(event.target.value);
+    }
+  }
+
+  /*
+   * Change state for the age when using AGE granularity, and update onset date
+   */
+  changeAge(event) {
+    const newDate = moment(this.props.birthdate).add(Number(event.target.value), this.state.ageUnit);
+
+    this.setState({
+      age: Number(event.target.value),
+      onsetDate: newDate,
+    });
+
+    if (this.props.onsetDateInputOnChange) {
+      this.props.onsetDateInputOnChange(newDate);
+    }
+  }
+
+  /*
+   * Change state for duration when using age granularity, and update onset date
+   */
+  changeAgeUnit(event) {
+    const newDate = moment(this.props.birthdate).add(1, event.target.value);
+
+    this.setState({
+      age: 1,
```

Comment:
Ah this was me trying to completely control the age value to ensure a invalid onsetDate was never calculated. I think that was wrong since the number field should be able to handle input validation and stop form submission. I'm going to remove this age reset and just let the numberField handle it.